File: src/main/java/de/tectoast/emolga/buttons/CalendarButton.java
package de.tectoast.emolga.buttons;

import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;

public class CalendarButton extends ButtonListener {
    public CalendarButton() {
        super("calendar");
    }

    @Override
    public void process(ButtonInteractionEvent e, String name) {
        if (name.equals("delete")) {
            e.reply("+1").setEphemeral(true).queue();
            e.getHook().deleteMessageById(e.getMessageId()).queue();
        }
    }
}

File: src/main/kotlin/de/tectoast/emolga/buttons/CalendarButton.kt
package de.tectoast.emolga.buttons

import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent

class CalendarButton : ButtonListener("calendar") {
    override suspend fun process(e: ButtonInteractionEvent, name: String) {
        if (name == "delete") {
            e.reply(":D").setEphemeral(true).queue()
            e.hook.deleteMessageById(e.messageId).queue()
        }
    }
}
